Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIText/UITextReleaseSystem.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.Throttling;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Optimization.Pools;
+using DCL.SDKComponents.SceneUI.Components;
+using DCL.SDKComponents.SceneUI.Groups;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.LifeCycle.Components;
+using UnityEngine.UIElements;
+
+namespace DCL.SDKComponents.SceneUI.Systems.UIText
+{
+    [UpdateInGroup(typeof(SyncedSimulationSystemGroup))]
+    [UpdateBefore(typeof(SceneUIComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.SCENE_UI)]
+    [ThrottlingEnabled]
+    public partial class UITextReleaseSystem : BaseUnityLoopSystem
+    {
+        private readonly IComponentPoolsRegistry poolsRegistry;
+
+        private UITextReleaseSystem(World world, IComponentPoolsRegistry poolsRegistry) : base(world)
+        {
+            this.poolsRegistry = poolsRegistry;
+        }
+
+        protected override void Update(float t)
+        {
+            HandleEntityDestructionQuery(World);
+            HandleUITextRemovalQuery(World);
+            World.Remove<UITextComponent>(in HandleUITextRemoval_QueryDescription);
+        }
+
+        [Query]
+        [None(typeof(PBUiText), typeof(DeleteEntityIntention))]
+        private void HandleUITextRemoval(ref UITextComponent uiTextComponent) =>
+            RemoveLabel(uiTextComponent);
+
+        [Query]
+        [All(typeof(DeleteEntityIntention))]
+        private void HandleEntityDestruction(ref UITextComponent uiTextComponent) =>
+            RemoveLabel(uiTextComponent);
+
+        private void RemoveLabel(UITextComponent uiTextComponent)
+        {
+            if (!poolsRegistry.TryGetPool(typeof(Label), out IComponentPool componentPool))
```

Comment:
You can store this pool in a field in `ctor` instead of resolving it every time: it's not meant to be changed at runtime

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/204/commits/a45ab5896a52e294383518914f7abcc20253cd45